1_fetch/src/unzip_file.py

```diff
@@ -0,0 +1,33 @@
+import os
+import zipfile
+
+def unzip_file(filename, source_dir, destination_dir):
+    """
+    Unzip file, save results to another directory.
+
+    :param filename: File to unzip, with full paths
+    :param source_dir: Directory to pull zipped file from.
+        This part of the path is removed from the beginning of filename to determine where to save unzipped files.
+    :param destination_dir: Directory to save unzipped files to.
+        In each file path, source_dir is replaced with destination_dir to determine where to save unzipped files.
+    :returns: list of paths to unzipped files
+
+    """
+    # check that file is inside source_dir
```

Comment:
I think this just checks that the path to `{filename}` is the same as `{source_dir}`, it doesn't check if `{filename}` actually exists in that `{source_dir}`. I tested this with `{filename}` equal to `'1_fetch/in/test.txt'` and `{source_dir}` set to `'1_fetch/in'`, and it passed the if statement, even though  `'1_fetch/in/test.txt'` doesn't exist
```python
filename = '1_fetch/in/test.txt'
source_dir = '1_fetch/in'

if os.path.samefile(source_dir, os.path.commonpath([filename, source_dir])):
    print(f'{filename} exists')
else:
    raise FileNotFoundError(f'File {f} not in directory {source_dir}')


> 1_fetch/in/test.txt exists
```

Reply:
Yes, good point! All it's checking is if the path to `filename` includes `source_dir`. 

I think that if the file itself doesn't exist, then `zipfile` will raise its own error. So, I don't think there's value in actually checking for the file itself. Instead, what if I update the comment and the error message to reflect the actual error checking that's happening here?

Reply:
that sounds good!